.azure-pipelines/scripts/RemoveSignatureScript.ps1

```diff
@@ -0,0 +1,30 @@
+function Remove-ThirdPartySignatures() {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $false)]
+        [string]$SigntoolPath,
+        [Parameter(Mandatory = $false)]
+        [string]$LayoutRoot)
+
+    $ErrorActionPreference = 'Stop'
+    $failedToUnsign = New-Object Collections.Generic.List[String]
+    $succesfullyUnsigned = New-Object Collections.Generic.List[String]
+    foreach ($tree in Get-ChildItem -Path "$LayoutRoot/bin" -Filter "*.dll" -Recurse | select FullName) {
+        try {
+            & "$SigntoolPath" remove /s /q "$($tree.FullName)" 2>&1
+            if ($lastexitcode -ne 0) {
+                $failedToUnsign.Add("$($tree.FullName)")
+            } else {
+                $succesfullyUnsigned.Add("$($tree.FullName)")
+            }
+        } catch {
+            $failedToUnsign.Add("$($tree.FullName)")
+        }
+    }
+    foreach ($f in $failedToUnsign) {
+        Write-Warning "Something went wrong, failed to process $f file in catch"
```

Comment:
Suppressing of the exception here would lead to the errors during agent validation later while self-updating - could it make sense to throw it here as error to give heads up?

Reply:
For now the script will exit with non zero status in case of any file for which we couldn't remove signature